src/main/webapp/profile.js

```diff
@@ -14,14 +14,19 @@
 
 function onProfileLoad() {
   prepareFormValidation();
+  supplyLogoutLink();
 }
 
-// Allows certain fields in the profile to be edited, hides edit button, and displays update button. 
-
-function onProfileLoad() {
-  supplyLogoutLink();
+// Submits profile form to Datastore.
+function submitProfileForm(methodType, redirectUrl) {
+  fetch(`/person?user-email=${$("#user-email").val()}&first-name=${$("#first-name-field").val()}`+
+    `&last-name=${$("#last-name-field").val()}&company=${$("#company-field").val()}&job=`+
+    `${$("#job-field").val()}&linkedin=${$("#linkedin-field").val()}`, 
+    {method: methodType});
+  window.location.replace(redirectUrl);
```

Comment:
This needs to only execute after the fetch goes through, so either add an await or .then promise capture.